src/balancerd/src/lib.rs

```diff
@@ -1047,8 +1063,16 @@ impl mz_server_core::Server for HttpsBalancer {
                     .map(|tenant| inner_metrics.tenant_connections(tenant));
 
                 let mut mz_stream = TcpStream::connect(resolved.addr).await?;
+
                 let mut client_counter = CountingConn::new(client_stream);
 
+                // Write the tcp proxy header
+                let addrs = ProxiedAddress::stream(peer_addr, resolved.addr);
+                let header = ProxyHeader::with_address(addrs);
+                let mut buf = [0u8; 1024];
+                let len = header.encode_to_slice_v2(&mut buf)?;
+                mz_stream.write_all(&buf[..len]).await?;
```

Comment:
This works, and I suspect it'll work even if we used MTLS connections to ENVD, but I do think it's an out of spec usage for proxy protocol headers which are generally meant to be applied on tcp connection and agnostic to underlying protocols.

I'm not super worried about this, the protocol should be agnostic to this, and is still incompatible with both http and pgwire requests. If we're not ok with this usage, I'll have to re-think some things.

Reply:
Not worried about strictly following the protocol here at all. What you have here seems great! We have full control over both the server and client and so who cares if we follow the protocol as long as both our server and client understand it.